Source/MLXNN/PositionalEncoding.swift

```diff
@@ -167,3 +167,50 @@ final public class ALiBi: Module {
         return attentionScores + alibiMask
     }
 }
+
+public class SuScaledRotaryEmbedding: Module {
+    let dimensions: Int
+    let base: Float
+    let maxPositionEmbeddings: Int
+    let originalMaxPositionEmbeddings: Int
+    let scale: Float
+    let _freqs: MLXArray
+
+    public init(
+        dimensions: Int,
+        base: Float = 10000.0,
+        maxPositionEmbeddings: Int = 131072,
+        originalMaxPositionEmbeddings: Int = 4096,
+        // !! shortFactor is not used in the new Python implementation
+        longFactor: [Float] = [1.0]
+    ) {
+        precondition(dimensions % 2 == 0, "Dimensions must be even")
+
+        self.dimensions = dimensions
+        self.base = base
+        self.maxPositionEmbeddings = maxPositionEmbeddings
+        self.originalMaxPositionEmbeddings = originalMaxPositionEmbeddings
+
+        let exponent =
+            MLXArray(stride(from: 0, to: dimensions, by: 2)).asType(.float32) / Float(dimensions)
+        let freqs = MLX.pow(MLXArray(base), exponent)
+        self._freqs = MLXArray(longFactor).asType(.float32) * freqs
+
+        self.scale = sqrt(
+            1 + log(Float(maxPositionEmbeddings) / Float(originalMaxPositionEmbeddings))
+                / log(Float(originalMaxPositionEmbeddings))
+        )
+    }
+
+    public func callAsFunction(_ x: MLXArray, offset: Int = 0) -> MLXArray {
+        return MLXFast.RoPE(
+            self.scale * x,
+            dimensions: x.shape.last!,
+            traditional: false,
+            base: self.base,  // !! Here the Python implementation passes in `None`, but this is not compatible with the Swift implementation of `MLXFast.RoPE`
+            scale: 1.0,
+            offset: offset
+                // !! Here the Python implementation passes self._freqs as argument `freqs` to `mx.fast.rope`, but this parameter does not exist in the Swift implementation of `MLXFast.RoPE`
```

Comment:
That API requires the latest MLX (0.17.0) which we haven't pulled into MLX Swift yet. Once we do, we should use the same API here as in Python.